### PR TITLE
Github actions improvements

### PR DESCRIPTION
**Pull request by FritzHoing**
This pull-request improves the existing github-workflows:

- enhances modularity by encapsulating a common operation into a reusable action for easier maintenance
- avoids duplicating the same logic across multiple jobs

----
📚 Documentation preview 📚: https://volto--6108.org.readthedocs.build/

### CODE REVIEW

**Comment by stevepiercy** [comment]
@FritzHoing this is a super useful contribution, but we can't accept it until you sign the [Plone Contributor Agreement](https://plone.org/foundation/contributors-agreement). This PR should also have a [change log entry](https://6.docs.plone.org/contributing/index.html#change-log-entry). Would you please take the necessary steps? Thank you!

**Comment by FritzHoing** [comment]
Thank you very much for your feedback! I signed the Contributor Agreement on monday already and just need to wait for the approval. The changelog entry is now added in the hope that the type is set correct for ci improvements.

**Comment by sneridagh** [comment]
@FritzHoing looks great! Thanks a lot for your contribution!

/cc @ericof

**Comment by sneridagh** [comment]
@FritzHoing I just allowed the tests to run, could you please take a look? It seems something is missing.

File: packages/volto/news/6108.internal
Comment (stevepiercy) [inline packages/volto/news/6108.internal:1]:
```suggestion
Improved the existing GitHub workflows by encapsulating a common operation into a reusable action for easier maintenance. @FritzHoing
```

**Review by stevepiercy — CHANGES_REQUESTED** [review]
This looks good, but tests are failing. I'll update it against `main` and see if that fixes it.

**Review by ericof — APPROVED** [review]

File: .github/actions/pre_test_setup/action.yml
Comment (ichim-david) [inline .github/actions/pre_test_setup/action.yml:50]:
```suggestion
name: pre test Setup

runs:
  using: "composite"
  steps:
    - uses: actions/checkout@v4

    - name: Use Node.js ${{ inputs.node-version }}
      uses: actions/setup-node@v4
      with:
        node-version: ${{ inputs.node-version }}

    - name: Enable corepack
      shell: bash
      run: corepack enable

    - name: Get pnpm store directory
      shell: bash
      run: |
        echo "STORE_PATH=$(pnpm store path --silent)" >> $GITHUB_ENV

    - uses: actions/cache@v4
      name: Setup pnpm cache
      with:
        path: ${{ env.STORE_PATH }}
        key: ${{ runner.os }}-pnpm-store-${{ hashFiles('**/pnpm-lock.yaml') }}
        restore-keys: |
          ${{ runner.os }}-pnpm-store-

    - name: Install dependencies
      shell: bash
      run: pnpm i

    - name: Cache Cypress Binary
      uses: actions/cache@v4
      with:
        path: ~/.cache/Cypress
        key: binary-${{ inputs.node-version }}-${{ hashFiles('pnpm-lock.yaml') }}

    - name: Install Cypress if not in cache
      if: steps.cache-cypress-binary.outputs.cache-hit != 'true'
      shell: bash
      working-directory: packages/volto
      run: make cypress-install

inputs:
  node-version:
    description: 'Node.js version'
    required: true
```

File: .github/workflows/acceptance.yml
Comment (ichim-david) [inline .github/workflows/acceptance.yml:17]:
```suggestion
      - uses: actions/checkout@v4

      - name: Pretest-Setup
        uses: ./.github/actions/pre_test_setup
        with:
          node-version: ${{ matrix.node-version }}
```

File: .github/workflows/acceptance.yml
Comment (ichim-david) [inline .github/workflows/acceptance.yml:61]:
```suggestion
      - uses: actions/checkout@v4

      - name: Pretest-Setup
        uses: ./.github/actions/pre_test_setup
        with:
          node-version: ${{ matrix.node-version }}
```

File: .github/workflows/acceptance.yml
Comment (ichim-david) [inline .github/workflows/acceptance.yml:105]:
```suggestion
      - uses: actions/checkout@v4

      - name: Pretest-Setup
        uses: ./.github/actions/pre_test_setup
        with:
          node-version: ${{ matrix.node-version }}
```

File: .github/workflows/acceptance.yml
Comment (ichim-david) [inline .github/workflows/acceptance.yml:149]:
```suggestion
      - uses: actions/checkout@v4

      - name: Pretest-Setup
        uses: ./.github/actions/pre_test_setup
        with:
          node-version: ${{ matrix.node-version }}
```

File: .github/workflows/acceptance.yml
Comment (ichim-david) [inline .github/workflows/acceptance.yml:193]:
```suggestion
      - uses: actions/checkout@v4

      - name: Pretest-Setup
        uses: ./.github/actions/pre_test_setup
        with:
          node-version: ${{ matrix.node-version }}
```

File: .github/workflows/acceptance.yml
Comment (ichim-david) [inline .github/workflows/acceptance.yml:237]:
```suggestion
      - uses: actions/checkout@v4

      - name: Pretest-Setup
        uses: ./.github/actions/pre_test_setup
        with:
          node-version: ${{ matrix.node-version }}
```

File: .github/workflows/acceptance.yml
Comment (ichim-david) [inline .github/workflows/acceptance.yml:280]:
```suggestion
      - uses: actions/checkout@v4

      - name: Pretest-Setup
        uses: ./.github/actions/pre_test_setup
        with:
          node-version: ${{ matrix.node-version }}
```

File: .github/workflows/acceptance.yml
Comment (ichim-david) [inline .github/workflows/acceptance.yml:324]:
```suggestion
      - uses: actions/checkout@v4

      - name: Pretest-Setup
        uses: ./.github/actions/pre_test_setup
        with:
          node-version: ${{ matrix.node-version }}
```

File: .github/workflows/acceptance.yml
Comment (ichim-david) [inline .github/workflows/acceptance.yml:436]:
```suggestion
      - uses: actions/checkout@v4

      - name: Pretest-Setup
        uses: ./.github/actions/pre_test_setup
        with:
          node-version: ${{ matrix.node-version }}
```

File: .github/workflows/acceptance.yml
Comment (ichim-david) [inline .github/workflows/acceptance.yml:481]:
```suggestion
      - uses: actions/checkout@v4

      - name: Pretest-Setup
        uses: ./.github/actions/pre_test_setup
        with:
          node-version: ${{ matrix.node-version }}
```

File: .github/workflows/acceptance.yml
Comment (ichim-david) [inline .github/workflows/acceptance.yml:691]:
```suggestion
      - uses: actions/checkout@v4

      - name: Pretest-Setup
        uses: ./.github/actions/pre_test_setup
        with:
          node-version: ${{ matrix.node-version }}
```

File: .github/workflows/acceptance.yml
Comment (ichim-david) [inline .github/workflows/acceptance.yml:737]:
```suggestion
      - uses: actions/checkout@v4

      - name: Pretest-Setup
        uses: ./.github/actions/pre_test_setup
        with:
          node-version: ${{ matrix.node-version }}
```

**Review by ichim-david — CHANGES_REQUESTED** [review]

**Review by ichim-david — APPROVED** [review]
I've commited the changes necessary to get the actions to work, tested and working.
I actually added a new pull request with the changes from this pull request plus my fixes but then I did the extra work to get commit suggestions which can commit to the parent as such I will close https://github.com/plone/volto/pull/6120

File: .github/workflows/acceptance.yml
Comment (stevepiercy) [inline .github/workflows/acceptance.yml:16]: This is duplicated at line 14.

```suggestion
```

File: .github/actions/pre_test_setup/action.yml
Comment (stevepiercy) [inline .github/actions/pre_test_setup/action.yml:14]: Is this necessary now? It wasn't there before. It makes sense to specify the shell when you need something that the shell provides, but not to just run a binary, as in this context.

File: .github/actions/pre_test_setup/action.yml
Comment (stevepiercy) [inline .github/actions/pre_test_setup/action.yml:30]: `Install dependencies` was previously after `Cache Cypress Binary`. Does that matter, or affect performance?

File: packages/volto/news/6108.internal
Comment (stevepiercy) [inline packages/volto/news/6108.internal:1]:
```suggestion
Improved the existing GitHub workflows by encapsulating a common operation into a reusable action for easier maintenance. @FritzHoing, @ichim-david
```

File: .github/actions/pre_test_setup/action.yml
Comment (stevepiercy) [inline .github/actions/pre_test_setup/action.yml:42]: Another possibly unnecessary `shell: bash`?

**Review by stevepiercy — CHANGES_REQUESTED** [review]
A couple questions, and some cleanup.

**Comment by ichim-david** [comment]
I made a branch out of the changes native to the volto repo in this pull request https://github.com/plone/volto/pull/6120
making it easier for me to handle change requests

**Comment by ichim-david** [comment]
Closing this pull request in favor of https://github.com/plone/volto/pull/6120 since having the pull request within Volto repo makes it easier to make changes as needed.

**Pull request closed without merging**
